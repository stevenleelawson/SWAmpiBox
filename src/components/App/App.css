@import url('https://fonts.googleapis.com/css?family=Aldrich');

.App {
  text-align: center;
}

.App-header {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: #ffe900;
  font-family: 'Aldrich', sans-serif;
  font-size: 5rem;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}
