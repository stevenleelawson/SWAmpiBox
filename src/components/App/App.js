import React, { Component } from 'react';
import OverViewCrawl from '../OverViewCrawl/OverViewCrawl';
import './App.css';
import ButtonContainer from '../ButtonContainer/ButtonContainer'

class App extends Component {
  constructor() {
    super();
    this.state = {
      starWars: []

    }
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Lawson's SwampiBox</h1>
        </header>

        <OverViewCrawl />
        <ButtonContainer />
      </div>
    );
  }
}

export default App;
