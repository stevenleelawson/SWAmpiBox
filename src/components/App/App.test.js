import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  let mockCard;
  let mockPeople;
  let getPeople;
  let fetchHomeworld;
  let fetchSpecies;
  let mockHomeWorld;
  beforeEach( () => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true})
    mockCard = {
      name: 'taco'
    }
    mockPeople = {
      results: [
        {name: 'Luke'}
      ]
    }
    mockHomeWorld = [ {name: 'Luke'}]
    // getPeople = jest.fn();
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('toggleFavorites should add a card to the favorites array', () => {
    wrapper.instance().toggleFavorites(mockCard);
    expect(wrapper.state('favorites').length).toEqual(1);
  })
  it('toggleFavorites should unfavorite a card', () => {
    wrapper.instance().toggleFavorites(mockCard);
    wrapper.instance().toggleFavorites(mockCard);
    expect(wrapper.state('favorites').length).toEqual(0);
  })
  it('should call fetch with the correct params', () => {
    window.fetch = jest.fn().mockImplementation(
      () => Promise.resolve({
        json: () => Promise.resolve(mockPeople)
      })
    )
    wrapper.instance().getPeople()
    expect(window.fetch).toHaveBeenCalledWith("https://swapi.co/api/people/")
  })
  it.skip('should update the people array in state', () => {
    const expected = {
      name: 'Luke'
    }
    window.fetch = jest.fn().mockImplementation(
      () => Promise.resolve({
        json: () => Promise.resolve(mockPeople)
      })
    )
    fetchSpecies = jest.fn().mockImplementation( () => Promise.resolve({
      mockPeople
    }))
    fetchHomeworld = jest.fn().mockImplementation( () => Promise.resolve({
      name: 'Luke'
    }))
    wrapper.instance().getPeople();
    wrapper.instance().fetchSpecies();
    wrapper.instance().fetchHomeworld();

    expect(wrapper.state('people')).toEqual(mockPeople)
  })
 })
