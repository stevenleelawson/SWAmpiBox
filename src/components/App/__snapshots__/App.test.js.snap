// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <Route
                    exact={true}
                    path="/favorites"
                    render={[Function]}
          />
          <h1
                    className="App-title"
          >
                    SWapiBox
          </h1>
</header>,
        <Route
          path="/"
          render={[Function]}
/>,
        <Route
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/people"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/vehicles"
          render={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
/>,
            <h1
              className="App-title"
>
              SWapiBox
</h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWapiBox",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "SWapiBox",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/people",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/vehicles",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
            />
            <h1
                        className="App-title"
            >
                        SWapiBox
            </h1>
</header>,
          <Route
            path="/"
            render={[Function]}
/>,
          <Route
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/people"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/vehicles"
            render={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
/>,
              <h1
                className="App-title"
>
                SWapiBox
</h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SWapiBox",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "SWapiBox",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/people",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/vehicles",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
